tests/subsys/bluetooth/mesh/scheduler_model/message_validity/src/main.c

```diff
@@ -0,0 +1,346 @@
+
+int bt_mesh_scene_srv_pub(struct bt_mesh_scene_srv *srv,
+			 struct bt_mesh_msg_ctx *ctx)
+{
+	return 0;
+}
+
+int bt_mesh_onoff_srv_pub(struct bt_mesh_onoff_srv *srv,
+			  struct bt_mesh_msg_ctx *ctx,
+			  const struct bt_mesh_onoff_status *status)
+{
+	return 0;
+}
+
+void bt_mesh_time_encode_time_params(struct net_buf_simple *buf,
+				     const struct bt_mesh_time_status *status)
+{
+}
+
+int bt_mesh_msg_send(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
+	       struct net_buf_simple *buf)
+{
+	return 0;
+}
+
+int bt_mesh_msg_ackd_send(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf, const struct bt_mesh_msg_rsp_ctx *rsp)
+{
+	return 0;
+}
+
+int _bt_mesh_time_srv_update_handler(struct bt_mesh_model *model)
+{
+	return 0;
+}
+
+static int action_set(const struct bt_mesh_schedule_entry *test_action,
+		       struct bt_mesh_model *sched_mod, bool ack)
+{
+	BT_MESH_MODEL_BUF_DEFINE(
+		buf, ack ? BT_MESH_SCHEDULER_OP_ACTION_SET : BT_MESH_SCHEDULER_OP_ACTION_SET_UNACK,
+		BT_MESH_SCHEDULER_MSG_LEN_ACTION_SET);
+
+	scheduler_action_pack(&buf, 0, test_action);
+
+	return _bt_mesh_scheduler_setup_srv_op[ack ? 0 : 1].func(sched_mod, NULL, &buf);
+}
+
+static int current_register_get(struct bt_mesh_model *sched_mod)
+{
+	BT_MESH_MODEL_BUF_DEFINE(buf, BT_MESH_SCHEDULER_OP_GET,
+				 BT_MESH_SCHEDULER_MSG_LEN_GET);
+
+	return _bt_mesh_scheduler_srv_op[0].func(sched_mod, NULL, &buf);
+}
+
+static int action_get(struct bt_mesh_model *sched_mod,
+				uint8_t idx)
+{
+	BT_MESH_MODEL_BUF_DEFINE(buf, BT_MESH_SCHEDULER_OP_ACTION_GET,
+				 BT_MESH_SCHEDULER_MSG_LEN_ACTION_GET);
+
+	net_buf_simple_add_u8(&buf, idx);
+
+	return _bt_mesh_scheduler_srv_op[1].func(sched_mod, NULL, &buf);
+}
+
+
+static void setup(void)
+{
+	mod_elem_idx_prep();
+
+	/* 1st of Jan 2010 */
+	start_tm.tm_year = 110;
+	start_tm.tm_mon = 0;
+	start_tm.tm_mday = 1;
+	start_tm.tm_wday = 5;
+	start_tm.tm_hour = 0;
+	start_tm.tm_min = 0;
+	start_tm.tm_sec = 0;
+
+	zassert_not_null(_bt_mesh_scheduler_srv_cb.init,
+			 "Init cb is null");
+
+	_bt_mesh_scheduler_srv_cb.init(sched_mod_elem);
+}
+
+static void teardown(void)
+{
+	zassert_not_null(_bt_mesh_scheduler_srv_cb.reset,
+			 "Reset cb is null");
+
+	_bt_mesh_scheduler_srv_cb.reset(sched_mod_elem);
+}
+
+
+static void test_set_prohibited(void)
+{
+	/** Attempt to send a SET / SET_UNACK message with a prohibited schedule time.
+	 * 1. Test that unacked set server message handler returns with error.
+	 * 2. Test that acked set server message handler returns with error.
+	 */
+
+	const struct bt_mesh_schedule_entry test_action = {
+		.year = 0x65, /* prohibited year */
+		.month = ANY_MONTH,
+		.day = 0,
+		.hour = 24,
+		.minute = 0,
+		.second = 30,
+		.day_of_week = ANY_DAY_OF_WEEK,
+		.action = BT_MESH_SCHEDULER_TURN_ON,
+		.transition_time = model_transition_encode(2500),
+		.scene_number = 1,
+	};
+
+	zassert_equal(action_set(&test_action, sched_mod_elem, false), -EINVAL);
```

Comment:
Also, I guess, test should check that nothing has been sent back to...

Reply:
Not sure if I understand. What did you want to be checked?